File: jar-executor-plugin/src/main/java/org/restler/KillProcessMojo.java
package org.restler;

import org.apache.maven.plugin.AbstractMojo;
import org.apache.maven.plugin.MojoExecutionException;
import org.apache.maven.plugin.MojoFailureException;
import org.apache.maven.plugins.annotations.LifecyclePhase;
import org.apache.maven.plugins.annotations.Mojo;

@Mojo(name = "stop", defaultPhase = LifecyclePhase.POST_INTEGRATION_TEST)
public class KillProcessMojo extends AbstractMojo {
    @Override
    public void execute() throws MojoExecutionException, MojoFailureException {
        getLog().info("Process is stopping.");
        CurrentProcess.killProcess();
    }
}

File: jar-executor-plugin/src/main/kotlin/org/restler/KillProcessMojo.kt
package org.restler

import org.apache.maven.plugin.AbstractMojo
import org.apache.maven.plugins.annotations.LifecyclePhase
import org.apache.maven.plugins.annotations.Mojo

@Mojo(name = "stop", defaultPhase = LifecyclePhase.POST_INTEGRATION_TEST)
class KillProcessMojo : AbstractMojo() {

    override fun execute() {
        log.info("Process is stopping.")
        CurrentProcess.killProcess()
    }

}
